posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,311 @@
+
+=== Downloading the Transformer 
+
+Several downloads are available from the project pages link:https://projects.eclipse.org/projects/technology.transformer/downloads[Downloads].
+
+Current snapshots are available at link:https://oss.sonatype.org/content/repositories/snapshots/org/eclipse/transformer/org.eclipse.transformer.cli/[Snapshots].
+
+=== Building the transformer
+
+Alternatively, the transformer can be built by cloning the link:https://github.com/eclipse/transformer[Eclipse Transformer repository] and invoking maven to build the transformer jar:
+
+[source]
+--
+    maven clean package
+--
+
+=== Preparing a work area
+
+Download or copy the transformer binaries to a local directory:
+
+[source]
+--
+    mkdir <work>
+    cd <work>
+    cp transformer/org.eclipse.transformer.cli/target/org.eclipse.transformer.cli-0.3.0-SNAPSHOT-distribution.jar .
+    unzip org.eclipse.transformer.cli-0.3.0-SNAPSHOT-distribution.jar
+--
+
+Unpackaging the distribution obtains:
+
+[source]
+--
+    org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar
+    libs/org.eclipse.transformer-0.3.0-SNAPSHOT.jar
+    libs/biz.aQute.bnd.transform-5.2.0.jar
+    libs/slf4j-api-1.7.25.jar
+    libs/commons-cli-1.4.jar
+    libs/slf4j-simple-1.7.30.jar
+--
+
+=== Running the transformer with default rules
+
+The transformer may now be run using the default rules:
+
+[source]
+--
+  java -jar org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar \
+    <target_location> \
+    <output_location>
+--
+
+(The jar 'org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar' is runnable, with the invocation selecting as main class 'org.eclipse.transformer.jakarta.JakartaTransformer' and with the necessary jars from the 'libs' folder included in the class path.)
+
+The transformer can be run using custom rules by specifying additional options on the command line.  The additional options specify local rules files that the transformer is to be used instead of the default rules files.
+
+For example, Open Liberty has more up-to-date rules for performing transformations.  To use the Open Liberty rules, download these from the Open Liberty GIT repository: link:https://github.com/OpenLiberty/open-liberty/tree/integration/dev/wlp-jakartaee-transform/rules[Open Liberty Rules].
+
+Add three command line options to the transformer command line invocation to pull in the more up-to-date rules data:
+
+[source]
+--
+  java -jar org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar \
+    <target_location> \
+    <output_location> \
+    -tr ${rules}/jakarta-renames.properties \
+    -td ${rules}/jakarta-direct.properties \
+    -tf ${rules}/jakarta-xml-master.properties
+--
+
+(These three command line options are most important.  While other command line options are available, for example, for transforming bundle archives, and for more finely selecting which target files are to be transformed, those other options are not usually needed.)
```

Comment:
The second sentence here is a run-on and needs to be reworded and probably needs to 2 or 3 sentences.

Reply:
I suggest possibly:  `Other command line options are available.  For example, there are options to transform bundle archives and finely selecting which target files are to be transformed.`

Reply:
I rewrote this to clarify things.